### PR TITLE
Added CPU Affinity support

### PR DESCRIPTION
**Pull request by DentOpt**
Changes to be committed:
	modified:   Os/CMakeLists.txt
	modified:   Os/Posix/TaskRoot.cpp

| | |
|:---|:---|
|**_Dentopt_**| |
|**_FW/OS_**|  |
|**_Linux_**|  |
|**_Taskroot.cpp deprecated - CPU affinity_**|  |
|**_Has Unit Tests (y)_**|  |
|**_Builds Without Errors yes_**|  |
|**_Unit Tests Pass - not validated_**|  |
|**_Documentation Included n _**|  |

---
## Change Description

A description of the changes contained in the PR.

## Rationale

To assign CPU affinities under Linux OS/Posix/TaskRoot.cpp needs to be used instead of Task.cpp

## Testing/Review Recommendations

Running tests on TaskRoot.cpp

## Future Work

Note any additional work that will be done relating to this issue.

### CODE REVIEW

File: Os/Posix/TaskRoot.cpp [hunk 1]
```diff
@@ -41,34 +41,10 @@ namespace Os {
 
         I32 stat = pthread_attr_init(&att);
         if (stat != 0) {
-            Fw::Logger::logMsg("pthread_attr_init: (%d)(%d): %s\n",stat,errno,strerror(stat));
+            Fw::Logger::logMsg("pthread_attr_init: (%d)(%d): %s\n",stat,errno, reinterpret_cast<POINTER_CAST>(strerror(stat)));
         	return TASK_INVALID_PARAMS;
         }
 
-        stat = pthread_attr_setstacksize(&att,stackSize);
```
Comment (LeStarch) [inline Os/Posix/TaskRoot.cpp:48]: Why were these lines removed?

Comment (DentOpt) [reply Os/Posix/TaskRoot.cpp:48]: `pthread_attr_setstacksize(&att,stackSize)` always returned 22 for me -> Led to segfault. 
Same for other attributed functions. 
I practically followed the setup of task.cpp where these attributes are not set for the linux configuration. 
However, I did not look into further details here.

Comment (timcanham) [reply Os/Posix/TaskRoot.cpp:48]: This was why we had a separate file for running as root, since many of these functions don't work in userspace. Can we keep the same partition in the file and add the CPU affinity functions to the regular `Task.cpp`?

**Review by timcanham — CHANGES_REQUESTED** [review]

**Comment by LeStarch** [comment]
This PR appears abandoned.  Closing.
